core/src/services/dropbox/core.rs

```diff
@@ -0,0 +1,149 @@
+use std::fmt::Formatter;
+use std::rc::Rc;
+
+use crate::raw::percent_encode_path;
+use crate::raw::HttpClient;
+use crate::raw::{new_json_deserialize_error, RpCopy};
+use crate::Error;
+use crate::ErrorKind;
+
+use bytes::Bytes;
+use http::request::Builder;
+use http::StatusCode;
+use http::{header, Request, Response};
+use serde::{Deserialize, Serialize};
+use tokio::sync::Mutex;
+
+use crate::{
+    raw::{new_request_build_error, AsyncBody, IncomingAsyncBody},
+    types::Result,
+};
+
+pub struct DropboxCore {
+    pub token: String,
+    pub client: HttpClient,
+    pub root: String,
+}
+
+impl Debug for DropboxCore {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut de = f.debug_struct("DropboxCore");
+        de.finish()
+    }
+}
+
+impl DropboxCore {
+    pub async fn dropbox_get(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url: String = "https://content.dropboxapi.com/2/files/download".to_string();
+        let download_args = DropboxDownloadArgs {
+            path: format!("{}{}", self.root, path).to_string(),
+        };
+        let request = self
+            .sign(Request::post(&url))
+            .header(
+                "Dropbox-API-Arg",
+                serde_json::to_string(&download_args).unwrap(),
+            )
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_update(
+        &self,
+        path: &str,
+        size: Option<usize>,
+        content_type: Option<&str>,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://content.dropboxapi.com/2/files/upload".to_string();
+        let mut args = DropboxUploadArgs::default();
+        args.path = format!("{}{}", self.root, path).to_string();
+        let mut request_builder = Request::post(&url);
+        if let Some(size) = size {
+            request_builder = request_builder.header(header::CONTENT_LENGTH, size);
+        }
+        if let Some(mime) = content_type {
+            request_builder = request_builder.header(header::CONTENT_TYPE, mime);
+        }
+        let request = self
+            .sign(request_builder)
+            .header("Dropbox-API-Arg", serde_json::to_string(&args).unwrap())
+            .body(body)
+            .map_err(new_request_build_error)?;
+
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_delete(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://api.dropboxapi.com/2/files/delete_v2".to_string();
+        let args = DropboxDeleteArgs {
+            path: format!("{}{}", self.root, path).to_string(),
+        };
+        let request = self
+            .sign(Request::post(&url))
+            .header(header::CONTENT_TYPE, "application/json")
+            .body(AsyncBody::Bytes(Bytes::from(
+                serde_json::to_string(&args).unwrap(),
```

Comment:
Please don't unwrap on `serde_json` calls, use [`new_json_serialize_error`](https://docs.rs/opendal/latest/opendal/raw/fn.new_json_serialize_error.html) to build the opendal error.